sno/working_copy/postgis.py

```diff
@@ -325,20 +239,25 @@ def delete_meta(self, dataset):
         """Delete any metadata that is only needed by this dataset."""
         pass  # There is no metadata except for the spatial_ref_sys table.
 
-    def _create_spatial_index(self, sess, dataset):
+    def _create_spatial_index_post(self, sess, dataset):
+        # Only implemented as _create_spatial_index_post:
+        # It is more efficient to write the features first, then index them all in bulk.
         L = logging.getLogger(f"{self.__class__.__qualname__}._create_spatial_index")
 
         geom_col = dataset.geom_column_name
+        index_name = f"{dataset.table_name}_idx_{geom_col}"
+        table = self._table_def_for_dataset(dataset)
 
-        # Create the PostGIS Spatial Index
         L.debug("Creating spatial index for %s.%s", dataset.table_name, geom_col)
         t0 = time.monotonic()
-        sess.execute(
-            f"""
-            CREATE INDEX "{dataset.table_name}_idx_{geom_col}"
-            ON {self.table_identifier(dataset)} USING GIST ({self.quote(geom_col)});
-            """
+
+        spatial_index = Index(
+            index_name, table.columns[geom_col], postgres_using="GIST"
         )
+        spatial_index.table = table
+        spatial_index.create(sess.connection())
+        sess.execute(f"""ANALYSE {self.table_identifier(dataset)};""")
```

Comment:
```suggestion
        sess.execute(f"""ANALYZE {self.table_identifier(dataset)};""")
```

kinda implies it's not tested though?

Reply:
Switched to ANALYZE

ANALYSE is apparently a synonym for ANALYZE, switching to any other misspelling causes tests to fail

Reply:
aha, someone added a secret alias for proper English? 👍